src/middleware/bodyParser/formData.ts

```diff
@@ -0,0 +1,89 @@
+import type { NextFunction, OpineRequest, OpineResponse } from "../../types.ts";
+import { hasBody, MultipartReader } from "../../../deps.ts";
+import { typeChecker } from "./typeChecker.ts";
+
+/**
+ * Create a middleware to parse formData.
+ *
+ * @param {object} [options]
+ * @return {function}
+ * @public
+ */
+export function formData(options: any = {}) {
+  const type = options.type || "multipart/form-data";
+
+  // create the appropriate type checking function
+  const shouldParse = typeof type !== "function" ? typeChecker(type) : type;
+
+  return async function formParser(
+    req: OpineRequest,
+    _res: OpineResponse,
+    next: NextFunction,
+  ) {
+    if (req._parsedBody) {
+      next();
+
+      return;
+    }
+
+    // skip requests without bodies
+    if (
+      !hasBody(req.headers) ||
+      parseInt(req.headers.get("content-length") || "") === 0
+    ) {
+      req.parsedBody = new FormData();
+      req.body = req.parsedBody;
+
+      next();
+
+      return;
+    }
+
+    // determine if request should be parsed
+    if (!shouldParse(req)) {
+      next();
+
+      return;
+    }
+
+    // read
+    const type = req.headers.get("Content-Type") || "";
+
+    const boundary = type.match(/boundary=(?:([^;]+))/);
+
+    if (!boundary) {
+      next();
+
+      return;
+    }
+
+    const multipartReader = new MultipartReader(req.raw, boundary.at(-1) || "");
+    const customFormData = new FormData();
+
+    const body = await multipartReader.readForm();
```

Comment:
I wonder if the effort of converting a std lib `MultipartFormData` to a `FormData` is worth it? `MultipartFormData` has everything a user could need in it's API (https://deno.land/std@0.134.0/mime/multipart.ts#L254) and removes the overhead of having to convert between these types and the care that is then needed with being performant and handling memory well?

Reply:
FormData gives way better control over the form rather than MultipartFormData, so I thought it would be better to convert multipartformdata to FormData